src/components/pages/dashboard/panels/map/mapPanel.js

```diff
@@ -33,62 +34,33 @@ const deviceToMapPin = ({ id, properties, type }) =>
 
 export class MapPanel extends Component {
 
-  componentDidMount() {
-    if (!this.map && this.props.azureMapsKey) {
-      this.initializeMap(this.props.azureMapsKey);
-    }
-    this.calculatePins(this.props, true);
-  }
-
   componentWillReceiveProps(nextProps) {
-    if (!this.map && nextProps.azureMapsKey) {
-      this.initializeMap(nextProps.azureMapsKey);
-    }
     this.calculatePins(nextProps);
   }
 
-  initializeMap(azureMapsKey) {
-    const center = new AzureMaps.data.Position(...Config.mapCenterPosition);
-    this.map = new AzureMaps.Map('map', {
-      'subscription-key': azureMapsKey,
-      center,
-      zoom: 11
-    });
-
+  onMapReady = (map) => {
+    this.map = map;
```

Comment:
Set this.map at the end of this function, just before the call to calculatePins.  That way there's no chance that map is there but the layers aren't when calculatePins is called from other places (like componentWillReceiveProps).

Reply:
Nice catch!